File: Csharp/CodeWars/src/Resolutions/EqualSidesOfAnArray.cs
namespace CodeWars.Resolutions;

public class EqualSidesOfAnArray
{
    public static int FindEvenIndex(int[] arr)
    {
        for (int i = 0; i < arr.Length; i++)
            if (CalculateArrayElements(arr, i)) return i;

        return -1;
    }

    private static bool CalculateArrayElements(int[] arr, int i)
    {
        var initialPart = arr[0..i].Sum();
        var finalPart = arr.Skip(i + 1).Sum();

        return initialPart == finalPart;
    }
}

File: TypeScript/Codewars/src/resolutions/Equal-Sides-Of-An-Array.ts
export function findEvenIndex(arr: number[]): number {
	for (let index = 0; index < arr.length; index++) {
		if (calculateArrayElements(arr, index)) return index;
	}

	return -1;
}

function calculateArrayElements(arr: number[], index: number): boolean {
	const initialPart = arr.slice(0, index).reduce((prev, curr) => prev + curr, 0);

	const finalPart = arr.slice(index + 1).reduce((prev, curr) => prev + curr, 0);

	return initialPart === finalPart;
}
